challenge4app/src/App.js

```diff
@@ -0,0 +1,169 @@
+  storageBucket: "contact-tracker-ccf9c.appspot.com",
+};
+firebase.initializeApp(config);
+
+// Get a reference to the database service
+  var database = firebase.database();
+
+var testCase = {
+  name: "nayanika",
+  name2: "caroline",
+}
+
+firebase.database().ref('testCase').set(testCase);
+var readData = firebase.database().ref("newContracts");
+
+class App extends Component {
+  constructor(props){
+    super(props);
+    this.state={
+      name:"",
+      description:"",
+      price:"",
+      index:"",
+      count: 0,
+      contracts:[]
+    }
+    this.handleNameChange=this.handleNameChange.bind(this);
+    this.handleDesriptionChange=this.handleDescriptionChange.bind(this);
+    this.handlePriceChange=this.handlePriceChange.bind(this);
+    this.handleIndexChange=this.handlePriceChange.bind(this);
+    this.handleSubmit=this.handleSubmit.bind(this);
+    //this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleNameChange(e){
+    this.setState({
+      ...this.state,
+      name:e.target.value
+    })
+    console.log(this.state)
+  }
+
+  handleDescriptionChange(e){
+    this.setState({
+      ...this.state,
+      description: e.target.value
+    })
+    console.log(this.state)
+  }
+
+  handlePriceChange(e){
+    this.setState({
+      ...this.state,
+      price: e.target.value
+    })
+    console.log(this.state)
+  }
+  
+
+    handleSubmit(e){
+      var infoArray=[];
+      infoArray.push(this.state.name);
+      infoArray.push(this.state.description);
+      infoArray.push(this.state.price);
+      infoArray.push(this.state.index); //info array ...retrieve the index by asking infoArray[3]; 
+      let newContracts = this.state.infoArray;
+      
+      let newContractsObject = {
+        ...this.state
+      }; 
+      
+      newContracts.push(newContractsObject);
+      
+      var count = this.state.count + 1;
+      
+      let index = {
+        if(index = null){
+            index = count;
+        }
+      }
+
+      this.setState({
+        ...this.state.newContracts  
+      })
+      console.log(this.state) //so here we need to NOT just print it...
+      e.preventDefault();
+    }
+
+/*
+    handleIndexChange(e){
+    //if statement to check if index already assigned; so would keep it when pressed submit. otherwise, increment.
+    const count = this.state.count +1; //attach that to onclick for submit //assign index on submit //need index field
+    const index = infoArray[3];
+    if(index != null){
+      //replace when submit is clicked (keep same index)
+      this.state.index = index; //do nothing
+    }
+    else{
+      //this.state.index = count; 
+      //infoArray.index = count;
+    }
+  }
+  */
+  //convert from array of arrays to an array of objects
+
+   /* 
+    handleClick() {
+    //find the index of the edit button -- listContracts  index 
+    //get the index by accessing the ... 
+    this.setState({
+      handleSubmit
+    })
+  }
+  */
```

Comment:
the beauty of git is you can always revert commits, especially if they are small and properly commented. commented out code has no place in a repository